### ISSUE TITLE
PostgreSQL migration issue from 21.x to 23.0.4

### ISSUE THREAD

**Issue by mweibel** [p2-inc/keycloak-orgs#154]
I tried to upgrade keycloak with the latest and greatest version (23.0.4) using your container image `quay.io/phasetwo/phasetwo-keycloak:23.0.4`. 

However I ran into the following issue:

```keycloak-1 keycloak 2024-01-23 16:19:14,679 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (production) mode
keycloak-1 keycloak 2024-01-23 16:19:14,679 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: liquibase.exception.ValidationFailedException: Validation Failed:
keycloak-1 keycloak      1 changesets check sum
keycloak-1 keycloak           META-INF/jpa-changelog-phasetwo-20200528.xml::1590682724768-crdb::garth (generated) was: 8:c4f727d3a9db8b66ceb9d4e6a268fa22 but is now: 8:d41d8cd98f00b204e9800998ecf8427e
keycloak-1 keycloak
keycloak-1 keycloak 2024-01-23 16:19:14,680 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Validation Failed:
keycloak-1 keycloak      1 changesets check sum
keycloak-1 keycloak           META-INF/jpa-changelog-phasetwo-20200528.xml::1590682724768-crdb::garth (generated) was: 8:c4f727d3a9db8b66ceb9d4e6a268fa22 but is now: 8:d41d8cd98f00b204e9800998ecf8427e
keycloak-1 keycloak
keycloak-1 keycloak 2024-01-23 16:19:14,680 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.
```

I'm opening this because I'm not quite sure how to resolve this. I assume the issue is coming from this line:
https://github.com/p2-inc/keycloak-orgs/blob/05583b89f629d6c465be4b6c129e23afcfd79ab4/src/main/resources/META-INF/jpa-changelog-phasetwo-20200528.xml#L21..L23

Any pointers on how to resolve this?

**Comment by xgp** [comment]
You can override the md5sum, but it shouldn't be necessary. Can you specify the version you are upgrading from? And the content of your `databasechangelog_organizati` table?

**Comment by xgp** [comment]
That migration is over 2 years old, which means if it was run successfully in `21.x`, it shouldn't be giving an error now.

**Comment by mweibel** [comment]
> That migration is over 2 years old, which means if it was run successfully in `21.x`, it shouldn't be giving an error now.

yeah that's what I saw too - very weird.

keycloak was at 21.1.2, orgs at 0.29-SNAPSHOT.jar.

table:
| id | author | filename | dateexecuted | orderexecuted | exectype | md5sum | description | comments | tag | liquibase | contexts | labels | deployment\_id |
| :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- | :--- |
| 1582221689607-1 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.503756 | 1 | EXECUTED | 8:dfe39f004e021d4a57fc2fe50266898b | createTable tableName=INVITATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-2 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.516056 | 2 | EXECUTED | 8:a04ab863adc4318c839d9454a9c29468 | createTable tableName=INVITATION\_TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-3 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.531085 | 3 | EXECUTED | 8:ecd274d4222aad2df9249f40af5fadba | createTable tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-4 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.545223 | 4 | EXECUTED | 8:1ec1dff0c81ebd4942d24dc61c05e854 | createTable tableName=ORGANIZATION\_ATTRIBUTE |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-5 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.555670 | 5 | EXECUTED | 8:7270b01f2cc6b3b31d01d8f7aa63b34b | createTable tableName=ORGANIZATION\_MEMBER |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-6 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.565712 | 6 | EXECUTED | 8:ea19a6e51483747c364d0fcb6fe7ff0d | createTable tableName=TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-7 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.582433 | 7 | EXECUTED | 8:9b14bbe203e8b8a8e53284eb9c0af77d | createTable tableName=TEAM\_ATTRIBUTE |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-8 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.593662 | 8 | EXECUTED | 8:e67c5deb714104348ffcf65b9602220d | createTable tableName=TEAM\_MEMBER |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-9 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.612718 | 9 | EXECUTED | 8:42a3e16bc5a761f8833b47fed917c194 | addPrimaryKey constraintName=INVITATIONPK, tableName=INVITATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-10 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.627311 | 10 | EXECUTED | 8:26f19702b99482faf2fb4bd76c476661 | addPrimaryKey constraintName=ORGANIZATIONPK, tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-11 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.640992 | 11 | EXECUTED | 8:e2a58f190b068c84aa1a9a2e8860b9c3 | addPrimaryKey constraintName=ORGANIZATION\_ATTRIBUTEPK, tableName=ORGANIZATION\_ATTRIBUTE |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-12 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.655325 | 12 | EXECUTED | 8:6fe4c7ad46a072975f8fef3ee883333b | addPrimaryKey constraintName=ORGANIZATION\_MEMBERPK, tableName=ORGANIZATION\_MEMBER |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-13 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.669240 | 13 | EXECUTED | 8:64bf08632be84bdae1857bb1e07ce74b | addPrimaryKey constraintName=TEAMPK, tableName=TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-14 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.683455 | 14 | EXECUTED | 8:4a94efe50d6453434b5d3a0cd3806e24 | addPrimaryKey constraintName=TEAM\_ATTRIBUTEPK, tableName=TEAM\_ATTRIBUTE |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-15 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.697579 | 15 | EXECUTED | 8:d298c550b954d2e7ccf3ef7214b8d22c | addPrimaryKey constraintName=TEAM\_MEMBERPK, tableName=TEAM\_MEMBER |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-16 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.712352 | 16 | EXECUTED | 8:c5edd7da6e089056fdd9aef70248ab14 | addUniqueConstraint constraintName=UK\_lhea5u4vqt3n9694kq7hyl5at, tableName=INVITATION\_TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-17 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.726514 | 17 | EXECUTED | 8:9626f361425ed3e2f6e07ab00f2df6e9 | addUniqueConstraint constraintName=UK\_lt3so1uv0pa7it6j7p9683oj4, tableName=INVITATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-18 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.744866 | 18 | EXECUTED | 8:dd72577e5ebd8143ce1a010760470ae6 | addUniqueConstraint constraintName=UK\_oht0oelmvemnqxwn3aj6maw6a, tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-19 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.758907 | 19 | EXECUTED | 8:05f9151597b8ff2608ee079efece3ac7 | addUniqueConstraint constraintName=UK\_th5mtjgw7jhxyjft8alqgdbiy, tableName=TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-20 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.773536 | 20 | EXECUTED | 8:769509d364d6f0d6af53454e666f1049 | addForeignKeyConstraint baseTableName=ORGANIZATION\_ATTRIBUTE, constraintName=FK\_519erdjtqivq189pm1ouanaix, referencedTableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-21 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.783613 | 21 | EXECUTED | 8:f2dafeccf22cdb19cda233d60e99754d | addForeignKeyConstraint baseTableName=TEAM\_MEMBER, constraintName=FK\_6ntjb7ds3m2cxgm08drxbcac8, referencedTableName=TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-22 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.798648 | 22 | EXECUTED | 8:4d3bd54abd6f0370c71c819eaf22ad8b | addForeignKeyConstraint baseTableName=TEAM\_ATTRIBUTE, constraintName=FK\_90avvlr99du3t58yefyi4c3jo, referencedTableName=TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-23 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.808034 | 23 | EXECUTED | 8:18e6227c9ac46ee482ede0f25d2aac69 | addForeignKeyConstraint baseTableName=ORGANIZATION\_MEMBER, constraintName=FK\_ci1r3v25slrartnlj0bd6wl0j, referencedTableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-24 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.818956 | 24 | EXECUTED | 8:31e4c43712ec7250253d185aa570ae9a | addForeignKeyConstraint baseTableName=INVITATION, constraintName=FK\_i4hnhwdvcobe55ohfwneyh6m4, referencedTableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-25 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.828496 | 25 | EXECUTED | 8:be4ba21c993e61e089a586c0c5912914 | addForeignKeyConstraint baseTableName=INVITATION\_TEAM, constraintName=FK\_lhea5u4vqt3n9694kq7hyl5at, referencedTableName=TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-26 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.837737 | 26 | EXECUTED | 8:c85d89b2d415edcff109197fd3bd8218 | addForeignKeyConstraint baseTableName=TEAM, constraintName=FK\_nfvcyxo3xogkp5go7057orvev, referencedTableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1582221689607-27 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200220.xml | 2023-04-05 09:22:54.846761 | 27 | EXECUTED | 8:e84cbe91dd8460eb6ca74ee5bd121495 | addForeignKeyConstraint baseTableName=INVITATION\_TEAM, constraintName=FK\_t6n7td7bgbuqyu9u29qxgli1e, referencedTableName=INVITATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-1 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.870254 | 28 | EXECUTED | 8:b86b33ec1add854e85d674754dc248a6 | createTable tableName=USER\_ORGANIZATION\_ROLE\_MAPPING |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-2 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.879150 | 29 | EXECUTED | 8:13e54807bf8aab3ff5b5fe2725850377 | addColumn tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-3 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.887648 | 30 | EXECUTED | 8:d8817b1a7e4934a8ea540140dcd37e6b | addColumn tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-4 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.896518 | 31 | EXECUTED | 8:8264e7d38b73f23b955bbdc1d7d5004b | addColumn tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-5 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.909576 | 32 | EXECUTED | 8:6a5c73bfb9bbc4948239cfb78496e37a | addUniqueConstraint constraintName=UK\_19A0TR48O23ALOR84GB8E3GC2, tableName=ORGANIZATION\_ATTRIBUTE |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-6 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.923307 | 33 | EXECUTED | 8:63b2674e1a638d6a2ff63345a7d08816 | addUniqueConstraint constraintName=UK\_4J97PTANUCPFF6O5H5PYMSONY, tableName=TEAM\_MEMBER |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-7 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.937732 | 34 | EXECUTED | 8:64d09299c5f6434fed0551b680881a87 | addUniqueConstraint constraintName=UK\_AU3GVA7EGXXE5C7V5N6VL9WBE, tableName=TEAM\_ATTRIBUTE |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-8 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.952187 | 35 | EXECUTED | 8:89b37cb6b8cf3860e416da8fc4ef312d | addUniqueConstraint constraintName=UK\_S5FNFVSRU9I4QPH9GB8V4FXFM, tableName=USER\_ORGANIZATION\_ROLE\_MAPPING |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-9 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.967623 | 36 | EXECUTED | 8:6acbe8a2e63f1dd25994363a540ac699 | addUniqueConstraint constraintName=UK\_SKS0GAK7WC152SFNO502TD78E, tableName=ORGANIZATION\_MEMBER |  | null | 4.16.1 | null | null | 0686574468 |
| 1587680343686-10 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200424.xml | 2023-04-05 09:22:54.977235 | 37 | EXECUTED | 8:446f61cf1452eb2e3429a8c224d4824e | addForeignKeyConstraint baseTableName=USER\_ORGANIZATION\_ROLE\_MAPPING, constraintName=FK\_HGF6S4UUNYWDKP4244YTNGBAD, referencedTableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| 1590682724768-2 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200528.xml | 2023-04-05 09:22:54.996439 | 38 | EXECUTED | 8:d346e29c6292be94ceefd1e9edf6c263 | dropPrimaryKey tableName=USER\_ORGANIZATION\_ROLE\_MAPPING |  | null | 4.16.1 | null | null | 0686574468 |
| 1590682724768-3 | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200528.xml | 2023-04-05 09:22:55.011726 | 39 | EXECUTED | 8:183479feff15409446c8a69ca74043ae | addPrimaryKey constraintName=USER\_ORGANIZATION\_ROLE\_MAPPINGPK, tableName=USER\_ORGANIZATION\_ROLE\_MAPPING |  | null | 4.16.1 | null | null | 0686574468 |
| 1590682724768-crdb | garth \(generated\) | META-INF/jpa-changelog-phasetwo-20200528.xml | 2023-04-05 09:22:55.017188 | 40 | EXECUTED | 8:c4f727d3a9db8b66ceb9d4e6a268fa22 | sql |  | null | 4.16.1 | null | null | 0686574468 |
| 20200611-0 | garth | META-INF/jpa-changelog-phasetwo-20200611.xml | 2023-04-05 09:22:55.025599 | 41 | EXECUTED | 8:0301a87566bb7a040ef7a19a18920e6e | addColumn tableName=TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 20200611-1 | garth | META-INF/jpa-changelog-phasetwo-20200611.xml | 2023-04-05 09:22:55.038866 | 42 | EXECUTED | 8:0fd6f20d316b3e65adbfec157a7168b2 | addColumn tableName=TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 20200611-1-0 | garth | META-INF/jpa-changelog-phasetwo-20200611-1.xml | 2023-04-05 09:22:55.048115 | 43 | EXECUTED | 8:8442cb390546382c88197a42bf598e11 | renameColumn newColumnName=TEAM\_ID, oldColumnName=teams\_ID, tableName=INVITATION\_TEAM |  | null | 4.16.1 | null | null | 0686574468 |
| 20200611-1- | garth | META-INF/jpa-changelog-phasetwo-20200611-1.xml | 2023-04-05 09:22:55.052870 | 44 | MARK\_RAN | 8:95050f3fd6bb85806e635a590c4bf164 | dropForeignKeyConstraint baseTableName=INVITATION\_TEAM, constraintName=FK\_lhea5u4vqt3n9694kq7hyl5at; dropUniqueConstraint constraintName=UK\_lhea5u4vqt3n9694kq7hyl5at, tableName=INVITATION\_TEAM; renameColumn newColumnName=TEAM\_ID, oldColumnName=tea... |  | null | 4.16.1 | null | null | 0686574468 |
| dropTable-example | liquibase-docs | META-INF/jpa-changelog-phasetwo-20211207.xml | 2023-04-05 09:22:55.098391 | 45 | MARK\_RAN | 8:761edebbcf14cade4b2a1784a6b998d4 | dropTable tableName=ORGANIZATION\_ROLE |  | null | 4.16.1 | null | null | 0686574468 |
| org-role-1 | garth | META-INF/jpa-changelog-phasetwo-20211207.xml | 2023-04-05 09:22:55.116231 | 46 | EXECUTED | 8:00df97f802d07a3a083cf3559afa6cbe | createTable tableName=ORGANIZATION\_ROLE |  | null | 4.16.1 | null | null | 0686574468 |
| org-role-2 | garth | META-INF/jpa-changelog-phasetwo-20211207.xml | 2023-04-05 09:22:55.126428 | 47 | EXECUTED | 8:fc822a61d8c1ae4edba68b9eb2c2b760 | addForeignKeyConstraint baseTableName=ORGANIZATION\_ROLE, constraintName=FK\_dsalfkmadslkfala4wf932f2w, referencedTableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| org-role-3 | garth | META-INF/jpa-changelog-phasetwo-20211207.xml | 2023-04-05 09:22:55.137137 | 48 | EXECUTED | 8:5705294033947ffded69964b5efb217b | addForeignKeyConstraint baseTableName=USER\_ORGANIZATION\_ROLE\_MAPPING, constraintName=FK\_123g9nf5findslfksadifu277, referencedTableName=ORGANIZATION\_ROLE |  | null | 4.16.1 | null | null | 0686574468 |
| drop-column-user-org-role-org-0 | garth | META-INF/jpa-changelog-phasetwo-20211208.xml | 2023-04-05 09:22:55.146747 | 49 | EXECUTED | 8:7776fbcd4917bc8d6110eedb5a638479 | dropUniqueConstraint constraintName=UK\_S5FNFVSRU9I4QPH9GB8V4FXFM, tableName=USER\_ORGANIZATION\_ROLE\_MAPPING |  | null | 4.16.1 | null | null | 0686574468 |
| drop-column-user-org-role-org-0-1 | garth | META-INF/jpa-changelog-phasetwo-20211208.xml | 2023-04-05 09:22:55.150470 | 50 | MARK\_RAN | 8:88e022286a20bc0a3c027f6c7d58e831 | dropForeignKeyConstraint baseTableName=USER\_ORGANIZATION\_ROLE\_MAPPING, constraintName=FK\_HGF6S4UUNYWDKP4244YTNGBAD; dropUniqueConstraint constraintName=UK\_S5FNFVSRU9I4QPH9GB8V4FXFM, tableName=USER\_ORGANIZATION\_ROLE\_MAPPING |  | null | 4.16.1 | null | null | 0686574468 |
| drop-column-user-org-role-org-1 | garth | META-INF/jpa-changelog-phasetwo-20211208.xml | 2023-04-05 09:22:55.163096 | 51 | EXECUTED | 8:158a23a5fe27b7638929f3cb63e1f0e8 | dropColumn tableName=USER\_ORGANIZATION\_ROLE\_MAPPING |  | null | 4.16.1 | null | null | 0686574468 |
| drop-column-user-org-role-org-2 | garth | META-INF/jpa-changelog-phasetwo-20211208.xml | 2023-04-05 09:22:55.187536 | 52 | EXECUTED | 8:fc74b394c35791adfee1395afc69f149 | addUniqueConstraint constraintName=UK\_S5FKFVSRU5I4QPHP8V8V4XXAM, tableName=USER\_ORGANIZATION\_ROLE\_MAPPING |  | null | 4.16.1 | null | null | 0686574468 |
| org-add-column-domain-0 | garth | META-INF/jpa-changelog-phasetwo-20211211.xml | 2023-04-05 09:22:55.198398 | 53 | EXECUTED | 8:b2a248774d03c4baf115e8c6b4577438 | addColumn tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| org-add-column-domain-0a | garth | META-INF/jpa-changelog-phasetwo-20211211.xml | 2023-04-05 09:22:55.213483 | 54 | EXECUTED | 8:d07ebb5f189b2381721cd0ae33aeeb56 | sql |  | null | 4.16.1 | null | null | 0686574468 |
| org-add-column-domain-1 | garth | META-INF/jpa-changelog-phasetwo-20211211.xml | 2023-04-05 09:22:55.229244 | 55 | EXECUTED | 8:f355cd9607a1f230bee148dca3700ad0 | addUniqueConstraint constraintName=UK\_th5m1423trnagf9dsloja05rm, tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| org-domain-unique-constraint-drop | garth | META-INF/jpa-changelog-phasetwo-20220131.xml | 2023-04-05 09:22:55.239863 | 56 | EXECUTED | 8:74ba5019af39c46232d8b9296bcaae98 | dropUniqueConstraint constraintName=UK\_th5m1423trnagf9dsloja05rm, tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| org-name-unique-constraint-drop | garth | META-INF/jpa-changelog-phasetwo-20220221.xml | 2023-04-05 09:22:55.251184 | 57 | EXECUTED | 8:5e13384695e60de8773a5f5d8d54a8ad | dropUniqueConstraint constraintName=UK\_19A0TR48O23ALOR84GB8E3GC2, tableName=ORGANIZATION\_ATTRIBUTE |  | null | 4.16.1 | null | null | 0686574468 |
| org-name-unique-constraint-drop-1 | garth | META-INF/jpa-changelog-phasetwo-20220221.xml | 2023-04-05 09:22:55.254819 | 58 | MARK\_RAN | 8:ee25cb31cf5ce3ca8bfe57b28fae14e6 | dropForeignKeyConstraint baseTableName=ORGANIZATION\_ATTRIBUTE, constraintName=FK\_519erdjtqivq189pm1ouanaix; dropUniqueConstraint constraintName=UK\_19A0TR48O23ALOR84GB8E3GC2, tableName=ORGANIZATION\_ATTRIBUTE |  | null | 4.16.1 | null | null | 0686574468 |
| org-role-add-column-description | garth | META-INF/jpa-changelog-phasetwo-20220322.xml | 2023-04-05 09:22:55.271309 | 59 | EXECUTED | 8:2b83476e960f04c925c8f4c45643ae8b | addColumn tableName=ORGANIZATION\_ROLE |  | null | 4.16.1 | null | null | 0686574468 |
| org-add-domain-table | garth | META-INF/jpa-changelog-phasetwo-20220322.xml | 2023-04-05 09:22:55.289025 | 60 | EXECUTED | 8:0880da128a679da43049c30d72ce4726 | dropColumn tableName=ORGANIZATION; createTable tableName=ORGANIZATION\_DOMAIN; addForeignKeyConstraint baseTableName=ORGANIZATION\_DOMAIN, constraintName=FK\_NYM2P7J4BST7GFC1DV8OAT264, referencedTableName=ORGANIZATION |  | null | 4.16.1 | null | null | 0686574468 |
| org-role-uq-constraint | garth | META-INF/jpa-changelog-phasetwo-20220628.xml | 2023-04-05 09:22:55.312777 | 61 | EXECUTED | 8:1c1941a4aeb55367fb7f4bf0892544ee | addUniqueConstraint constraintName=UK\_N6IFCXQHTNVMANYP0QIXDAO9G, tableName=ORGANIZATION\_ROLE |  | null | 4.16.1 | null | null | 0686574468 |
| invitation-add-role-table | garth | META-INF/jpa-changelog-phasetwo-20220824.xml | 2023-04-05 09:22:55.325302 | 62 | EXECUTED | 8:b456ac1f9c251ac0ac51eb9e29f75620 | createTable tableName=INVITATION\_ROLE; addForeignKeyConstraint baseTableName=INVITATION\_ROLE, constraintName=FK\_R8W81028FYO57BQ4CTDJFZSYP, referencedTableName=INVITATION |  | null | 4.16.1 | null | null | 0686574468 |
| domain-add-id | garth | META-INF/jpa-changelog-phasetwo-20220911.xml | 2023-04-05 09:22:55.345992 | 63 | EXECUTED | 8:ee7a795e20997a74291ead9ea4459d26 | addColumn tableName=ORGANIZATION\_DOMAIN; addPrimaryKey constraintName=ORGANIZAT\_DOMAINPK, tableName=ORGANIZATION\_DOMAIN |  | null | 4.16.1 | null | null | 0686574468 |
| drop-UK-S5FNFVSRU9I4QPH9GB8V4FXFM-unique-constraint | grinay | META-INF/jpa-changelog-phasetwo-20211206.xml | 2023-04-28 08:45:37.992767 | 64 | MARK\_RAN | 8:aa06bded277a7c03b52b2f238f5fa421 | sql |  | null | 4.16.1 | null | null | 2671537874 |
| update-ROLE\_ID-to-VARCHAR-36 | grinay | META-INF/jpa-changelog-phasetwo-20211206.xml | 2023-04-28 08:45:38.016146 | 65 | MARK\_RAN | 8:a31e96b506d3883020754c8fb0ab4b11 | sql |  | null | 4.16.1 | null | null | 2671537874 |
| add-UK-S5FNFVSRU9I4QPH9GB8V4FXFM-unique-constraint | grinay | META-INF/jpa-changelog-phasetwo-20211206.xml | 2023-04-28 08:45:38.024286 | 66 | MARK\_RAN | 8:a6421bee2e98e0b0bfa144fce87d398b | sql |  | null | 4.16.1 | null | null | 2671537874 |
| drop-foreign-key-constraint | grinay | META-INF/jpa-changelog-phasetwo-20211207-1.xml | 2023-04-28 08:45:38.032499 | 67 | MARK\_RAN | 8:cc206673bcee53b1dce7d15093df4e7d | dropForeignKeyConstraint baseTableName=USER\_ORGANIZATION\_ROLE\_MAPPING, constraintName=FK\_HGF6S4UUNYWDKP4244YTNGBAD |  | null | 4.16.1 | null | null | 2671537874 |
| modify-organization-columns | grinay | META-INF/jpa-changelog-phasetwo-20230404.xml | 2023-04-28 08:45:38.049076 | 68 | MARK\_RAN | 8:ba0df0a315e312fe76c17e3825798be6 | modifyDataType columnName=DISPLAY\_NAME, tableName=ORGANIZATION; modifyDataType columnName=URL, tableName=ORGANIZATION |  | null | 4.16.1 | null | null | 2671537874 |

**Comment by xgp** [comment]
@mweibel I can recreate for that version pair, but not others. Still trying to figure out what's going on.

**Comment by xgp** [comment]
Would those of you who have experienced this in a repeatable fashion try this branch (https://github.com/p2-inc/keycloak-orgs/tree/xgp/liquibase-checksum) to see if it works without error? I added both checksums to liquibase's `validCheckSum` tag in that `changeSet`.

**Comment by mweibel** [comment]
Thanks - I'll test that with a local install (test/prod systems were already upgraded with the md5sum change workaround). Might take a few days though.

**Comment by awoimbee** [comment]
I had the same issue upgrading from 22.0.4 to 23, fixed by running an SQL UPDATE overriding the md5sum (no big deal).

**Comment by xgp** [comment]
Thanks. @awoimbee . It's good to have a workaround, but we'd prefer upgrades be seamless. That's why we're trying to build this into the library.

**Comment by patrickmichalina** [comment]
```sql
UPDATE databasechangelog_organizati
SET md5sum = '8:d41d8cd98f00b204e9800998ecf8427e'
WHERE md5sum = '8:c4f727d3a9db8b66ceb9d4e6a268fa22';
```

**Comment by xgp** [comment]
Fixed by #164
